src/index.js

```diff
@@ -0,0 +1,223 @@
+  const landscapeElement = document.getElementById("landscape");
+  tempValueElement.textContent = `${state.currentTemp}°`;
+
+  if (state.currentTemp >= 80) {
+    tempValueElement.style.color = "red";
+    landscapeElement.textContent = "🌵  🐍 🦂 🌵🌵  🐍 🏜 🦂";
+  } else if (state.currentTemp >= 70) {
+    tempValueElement.style.color = "orange";
+    landscapeElement.textContent = "🌸🌿🌼  🌷🌻🌿 ☘️🌱 🌻🌷";
+  } else if (state.currentTemp >= 60) {
+    tempValueElement.style.color = "salmon";
+    landscapeElement.textContent = "🌾🌾 🍃 🪨  🛤 🌾🌾🌾 🍃";
+  } else if (state.currentTemp >= 50) {
+    tempValueElement.style.color = "green";
+    landscapeElement.textContent = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+  } else if (state.currentTemp <= 49) {
+    tempValueElement.style.color = "blue";
+    landscapeElement.textContent = "⛄️ ⛄️ ⛄️";
+  }
+};
+
+// Function to increase temperature
+const increaseTemp = () => {
+  state.currentTemp += 1;
+  updateTemperatureDisplay();
+};
+
+// Function to decrease temperature
+const decreaseTemp = () => {
+  state.currentTemp -= 1;
+  updateTemperatureDisplay();
+};
+
+/** Wave 2 events **/
+const temperatureControls = () => {
+  const warmUpButton = document.getElementById("increaseTempControl");
+  const coolDownButton = document.getElementById("decreaseTempControl");
+
+  warmUpButton.addEventListener("click", increaseTemp);
+  coolDownButton.addEventListener("click", decreaseTemp);
+};
+
+
+/************************/
+/******* Wave 3 *********/
+/************************/
+
+const updateCity = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const currentCityHeader = document.getElementById("headerCityName");
+  currentCityHeader.textContent = cityNameInput.value;
+};
+
+const handleUpdateCityEvent = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const resetButton = document.getElementById("cityNameReset");
+  cityNameInput.addEventListener("input", updateCity);
+  resetButton.addEventListener("click", ResetCity);
+};
+  
+/************************/
+/******* Wave 4 *********/
+/************************/
+
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+// Event handler function to retrieve user input:
+const retrieveCityInput = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  return cityNameInput.value;
+};
+
+// API Call to LocationIQ for location coordinates
+const getCoordinates = () => {
+  let location = retrieveCityInput();
+
+  return axios
+    .get("http://127.0.0.1:5000/location", {
+      params: {
+        q: location,
+      },
+    })
+    .then((response) => {
+      let latitude = response.data[0].lat;
+      let longitude = response.data[0].lon;
+
+      console.log(
+        `Location: ${location}, Latitude: ${latitude}, Longitude: ${longitude}`
+      );
+      return {
+        latitude: latitude,
+        longitude: longitude,
+      };
+    })
```

Comment:
Great work on this promise! Very easy to follow! Especially since you also decided to use different helper functions to tie things together!